src/agent/holiday_hours.rs

```diff
@@ -0,0 +1,376 @@
+//! Holiday hours metadata parsing and evaluation logic
+
+use {
+    anyhow::{
+        anyhow,
+        Context,
+        Result,
+    },
+    chrono::{
+        naive::NaiveTime,
+        DateTime,
+        Datelike,
+        Duration,
+        Utc,
+    },
+    chrono_tz::Tz,
+    lazy_static::lazy_static,
+    std::str::FromStr,
+};
+
+lazy_static! {
+    /// Helper time value representing 24:00:00 as 00:00:00 minus 1
+    /// nanosecond (underflowing to 23:59:59.999(...) ). While chrono
+    /// has this value internally exposed as NaiveTime::MAX, it is not
+    /// exposed outside the crate.
+    static ref MAX_TIME_INSTANT: NaiveTime = NaiveTime::MIN.overflowing_sub_signed(Duration::nanoseconds(1)).0;
+}
+
+/// Holiday hours schedule
+#[derive(Clone, Default, Debug, Eq, PartialEq)]
+pub struct HolidaySchedule {
+    pub timezone: Option<Tz>,
+    pub days:     Vec<HolidayDaySchedule>,
+}
+
+impl HolidaySchedule {
+    pub fn all_closed() -> Self {
+        Self {
+            timezone: Default::default(),
+            days:     vec![],
+        }
+    }
+
+    pub fn can_publish_at(&self, when: DateTime<Utc>) -> bool {
+        // Convert to time local to the market
+        let when_local = when.with_timezone(&self.timezone.unwrap());
+
+        let market_month = when_local.date_naive().month0() + 1;
+        let market_day = when_local.date_naive().day0() + 1;
+
+        let market_time = when_local.time();
+
+        for day in &self.days {
+            // Check if the day matches
+            if day.month == market_month && day.day == market_day {
+                return day.kind.can_publish_at(market_time);
+            }
+        }
+        true
+    }
+}
+
+impl FromStr for HolidaySchedule {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self> {
+        let split_by_commas = s.split(",");
+        let mut days = Vec::new();
+
+        for day_str in split_by_commas {
+            let day = day_str.parse()?;
+            days.push(day);
+        }
+
+        Ok(HolidaySchedule {
+            days,
+            timezone: None,
+        })
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct HolidayDaySchedule {
+    pub month: u32,
+    pub day:   u32,
+    pub kind:  HolidayDayKind,
+}
+
+impl FromStr for HolidayDaySchedule {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self> {
+        let date_time_parts: Vec<&str> = s.split("/").collect();
+        if date_time_parts.len() != 2 {
+            return Err(anyhow!("Invalid format"));
+        }
+
+        if date_time_parts[0].len() != 4 {
+            return Err(anyhow!("Invalid date format"));
+        }
+
+        let month: u32 = date_time_parts[0][..2]
+            .parse()
+            .map_err(|_| anyhow!("Invalid month"))?;
+        let day: u32 = date_time_parts[0][2..]
+            .parse()
+            .map_err(|_| anyhow!("Invalid day"))?;
+        let kind: HolidayDayKind = date_time_parts[1].parse()?;
```

Comment:
I don't think you are validating the day/month here? Should restrict these to day/month ranges. You also have a problem that it will depend on the month what day range is valid.

I think there is an assumption that the input string is trusted, if so can you comment that?

Reply:
comment added.